typescript/sdk/src/worker.ts

```diff
@@ -0,0 +1,216 @@
+import { aptos } from "@aptos-labs/aptos-indexer-protos";
+import { Config } from "./config";
+import { Timer } from "timer-node";
+import { exit } from "process";
+import { ChannelCredentials, Metadata } from "@grpc/grpc-js";
+import { parse as pgConnParse } from "pg-connection-string";
+import { createDataSource } from "./models/data_source";
+import {
+  NextVersionToProcess,
+  createNextVersionToProcess,
+} from "./models/next_version_to_process";
+import { Base } from "./models/base";
+import { TransactionsProcessor } from "./processor";
+import { DataSource } from "typeorm";
+
+/// This worker class is responsible for connecting to the txn stream and dispatching
+/// transactions to the provided processor. It is responsible for tracking which
+/// version it has processed up to and checkpointing that version in the database.
+export class Worker {
+  // Base configuration required to connect to the database and the data stream.
+  config: Config;
+
+  // The processor we'll run.
+  processor: TransactionsProcessor;
+
+  // The DB connection.
+  dataSource: DataSource;
+
+  constructor({
```

Comment:
This is a pretty good start! Off the top of my head, it's missing a couple of key features: multithreading, GRPC reconnection, checking for gaps in txn version. We should add these before publishing the SDK.